avocado/utils/process.py

```diff
@@ -310,6 +312,12 @@ def __init__(self, command="", stdout=b"", stderr=b"",
         self.pid = pid
         if encoding is None:
             encoding = sys.getdefaultencoding()
+            if PY2 and encoding == "ascii":
```

Comment:
@clebergnu here I kept the "encoding" check, because ony CPython it's hardcoded, but some non-standard python2-like interpreter might define it differently so I think it makes sense to only set it if we detect this default. Hopefully soon we'll have a unified way and then we'll use that rather than this handling.